### PR TITLE
feat: add gem subcommands

### PR DESCRIPTION
**Pull request by AlanSyue**
**What kind of change does this PR introduce? (Bug fix, feature, docs update, ...)**
feature

**What is the current behavior? (You can also link to an open issue here)**
N/A

**What is the new behavior (if this is a feature change)?**
Add more gem subcommands

**Additional info:**
According to the [official document](https://guides.rubygems.org/command-reference/), GEM has the following commands:

- [x] [gem build](https://guides.rubygems.org/command-reference/#gem-build)
- [x] [gem cert](https://guides.rubygems.org/command-reference/#gem-cert)
- [x] [gem check](https://guides.rubygems.org/command-reference/#gem-check)
- [x] [gem cleanup](https://guides.rubygems.org/command-reference/#gem-cleanup)
- [x] [gem contents](https://guides.rubygems.org/command-reference/#gem-contents)
- [x] [gem dependency](https://guides.rubygems.org/command-reference/#gem-dependency)
- [x] [gem environment](https://guides.rubygems.org/command-reference/#gem-environment)
- [x] [gem fetch](https://guides.rubygems.org/command-reference/#gem-fetch)
- [x] [gem generate_index](https://guides.rubygems.org/command-reference/#gem-generate_index)
- [x] [gem help](https://guides.rubygems.org/command-reference/#gem-help)
- [x] [gem info](https://guides.rubygems.org/command-reference/#gem-info)
- [x] [gem install](https://guides.rubygems.org/command-reference/#gem-install)
- [x] [gem list](https://guides.rubygems.org/command-reference/#gem-list)
- [x] [gem lock](https://guides.rubygems.org/command-reference/#gem-lock)
- [x] [gem mirror](https://guides.rubygems.org/command-reference/#gem-mirror)
- [x] [gem open](https://guides.rubygems.org/command-reference/#gem-open)
- [x] [gem outdated](https://guides.rubygems.org/command-reference/#gem-outdated)
- [x] [gem owner](https://guides.rubygems.org/command-reference/#gem-owner)
- [x] [gem pristine](https://guides.rubygems.org/command-reference/#gem-pristine)
- [x] [gem push](https://guides.rubygems.org/command-reference/#gem-push)
- [x] [gem query](https://guides.rubygems.org/command-reference/#gem-query)
- [x] [gem rdoc](https://guides.rubygems.org/command-reference/#gem-rdoc)
- [x] [gem search](https://guides.rubygems.org/command-reference/#gem-search)
- [x] [gem server](https://guides.rubygems.org/command-reference/#gem-server)
- [x] [gem signin](https://guides.rubygems.org/command-reference/#gem-signin)
- [x] [gem signout](https://guides.rubygems.org/command-reference/#gem-signout)
- [x] [gem sources](https://guides.rubygems.org/command-reference/#gem-sources)
- [x] [gem specification](https://guides.rubygems.org/command-reference/#gem-specification)
- [x] [gem stale](https://guides.rubygems.org/command-reference/#gem-stale)
- [x] [gem uninstall](https://guides.rubygems.org/command-reference/#gem-uninstall)
- [x] [gem unpack](https://guides.rubygems.org/command-reference/#gem-unpack)
- [x] [gem update](https://guides.rubygems.org/command-reference/#gem-update)
- [x] [gem which](https://guides.rubygems.org/command-reference/#gem-which)
- [x] [gem yank](https://guides.rubygems.org/command-reference/#gem-yank)

### CODE REVIEW

**Comment by AlanSyue** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by AlanSyue** [comment]
@SeparateRecords Should we allow some subcommands to be released first? lol

File: src/gem.ts [hunk 1]
```diff
@@ -16,18 +16,1264 @@ const gems: Fig.Generator = {
...
+          name: ["-s", "--spec-dir"],
+          description: "Search for gems under specific paths",
+          args: {
+            name: "DIR",
+            template: "filepaths",
+          },
+        },
+        {
+          name: ["-l", "--lib-only"],
+          description: "Only return files in the Gem’s lib_dirs",
```
Comment (fedeci) [inline src/gem.ts:395]:
```suggestion
          description: "Only return files in the Gem’s lib_dirs",
          exclusiveOn: ["--no-lib-only"],
```

File: src/gem.ts [hunk 2]
```diff
@@ -16,18 +16,1264 @@ const gems: Fig.Generator = {
...
+        },
+        {
+          name: "--no-lib-only",
+          description: "Not only return files in the Gem’s lib_dirs",
+        },
+        {
+          name: "--prefix",
+          description: "Don’t include installed path prefix",
```
Comment (fedeci) [inline src/gem.ts:403]:
```suggestion
          description: "Don’t include installed path prefix",
          exclusiveOn: ["--no-prefix"],
```

File: src/gem.ts [hunk 3]
```diff
@@ -16,18 +16,1264 @@ const gems: Fig.Generator = {
...
+        },
+        {
+          name: "--no-prefix",
+          description: "Include installed path prefix",
```
Comment (fedeci) [inline src/gem.ts:407]:
```suggestion
          description: "Include installed path prefix",
          exclusiveOn: ["--prefix"],
```

File: src/gem.ts [hunk 4]
```diff
@@ -16,18 +16,1264 @@ const gems: Fig.Generator = {
...
+        },
+        {
+          name: "--show-install-dir",
+          description: "Show only the gem install dir",
```
Comment (fedeci) [inline src/gem.ts:411]:
```suggestion
          description: "Show only the gem install dir",
          exclusiveOn: ["--no-show-install-dir"],
```

File: src/gem.ts [hunk 5]
```diff
@@ -16,18 +16,1264 @@ const gems: Fig.Generator = {
...
+        },
+        {
+          name: "--no-show-install-dir",
+          description: "Do not show only the gem install dir",
```
Comment (fedeci) [inline src/gem.ts:415]:
```suggestion
          description: "Do not show only the gem install dir",
          exclusiveOn: ["--show-install-dir"],
```

File: src/gem.ts [hunk 6]
```diff
@@ -16,18 +16,1264 @@ const gems: Fig.Generator = {
...
+          name: "--platform",
+          description: "Specify the platform of gem to dependency",
+          args: {
+            name: "PLATFORM",
+            description: "The platform of gem",
+          },
+        },
+        {
+          name: "--prerelease",
+          description: "Allow prerelease versions of a gem",
```
Comment (fedeci) [inline src/gem.ts:445]:
```suggestion
          description: "Allow prerelease versions of a gem",
          exclusiveOn: ["--no-prerelease"],
```

File: src/gem.ts [hunk 7]
```diff
@@ -16,18 +16,1264 @@ const gems: Fig.Generator = {
...
+        },
+        {
+          name: "--no-prerelease",
+          description: "Do not allow prerelease versions of a gem",
```
Comment (fedeci) [inline src/gem.ts:449]:
```suggestion
          description: "Do not allow prerelease versions of a gem",
          exclusiveOn: ["--prerelease"],
```

File: src/gem.ts [hunk 8]
```diff
@@ -16,18 +16,1264 @@ const gems: Fig.Generator = {
...
+        },
+        {
+          name: ["-R", "--reverse-dependencies"],
+          description: "Include reverse dependencies in the output",
```
Comment (fedeci) [inline src/gem.ts:453]:
```suggestion
          description: "Include reverse dependencies in the output",
          exclusiveOn: ["--no-reverse-dependencies"],
```

File: src/gem.ts [hunk 9]
```diff
@@ -16,18 +16,1264 @@ const gems: Fig.Generator = {
...
+        },
+        {
+          name: "--no-reverse-dependencies",
+          description: "Do not include reverse dependencies in the output",
```
Comment (fedeci) [inline src/gem.ts:457]:
```suggestion
          description: "Do not include reverse dependencies in the output",
          exclusiveOn: ["--reverse-dependencies"],
```

File: src/gem.ts [hunk 10]
```diff
@@ -213,6 +1459,56 @@ const completionSpec: Fig.Spec = {
...
+      isPersistent: true,
+    },
+    {
+      name: ["-V", "--verbose"],
+      description: "Set the verbose level of output",
+      isPersistent: true,
+    },
+    {
+      name: "--no-verbose",
+      description: "Do not set the verbose level of output",
```
Comment (fedeci) [inline src/gem.ts:1471]:
```suggestion
      description: "Do not set the verbose level of output",
      exclusiveOn: ["--verbose"],
```

File: src/gem.ts [hunk 11]
```diff
@@ -213,6 +1459,56 @@ const completionSpec: Fig.Spec = {
     {
       name: ["--help", "-h"],
       description: "Show help for gem",
+      isPersistent: true,
+    },
+    {
+      name: ["-V", "--verbose"],
+      description: "Set the verbose level of output",
+      isPersistent: true,
```
Comment (fedeci) [inline src/gem.ts:1467]:
```suggestion
      isPersistent: true,
      exclusiveOn: ["--no-verbose"],
```

File: src/gem.ts [hunk 12]
```diff
@@ -16,18 +16,1264 @@ const gems: Fig.Generator = {
...
+      description: "Display local gems whose name matches REGEXP",
+      args: {
+        name: "REGEXP",
+        description: "Regexp to look for in gem name",
+        isOptional: true,
+      },
+      options: [
+        {
+          name: ["-d", "--details"],
+          description: "Display detailed information of gem(s)",
```
Comment (fedeci) [inline src/gem.ts:1154]:
```suggestion
          description: "Display detailed information of gem(s)",
          exclusiveOn: ["--no-details"],
```

File: src/gem.ts [hunk 13]
```diff
@@ -16,18 +16,1264 @@ const gems: Fig.Generator = {
...
+        },
+        {
+          name: "--no-details",
+          description: "Do not display detailed information of gem(s)",
```
Comment (fedeci) [inline src/gem.ts:1158]:
```suggestion
          description: "Do not display detailed information of gem(s)",
          exclusiveOn: ["--details"],
```

File: src/gem.ts [hunk 14]
```diff
@@ -16,18 +16,1264 @@ const gems: Fig.Generator = {
...
+        },
+        {
+          name: ["-u", "--update-sources"],
+          description: "[Deprecated] Update local source cache",
```
Comment (fedeci) [inline src/gem.ts:1162]:
```suggestion
          description: "[Deprecated] Update local source cache",
          deprecated: true,
          exclusiveOn: ["--no-update-sources"],
```

File: src/gem.ts [hunk 15]
```diff
@@ -16,18 +16,1264 @@ const gems: Fig.Generator = {
...
+        },
+        {
+          name: "--no-update-sources",
+          description: "[Deprecated] Do not update local source cache",
```
Comment (fedeci) [inline src/gem.ts:1166]:
```suggestion
          description: "[Deprecated] Do not update local source cache",
          deprecated: true,
          exclusiveOn: ["--update-sources"],
```

File: src/gem.ts [hunk 16]
```diff
@@ -16,18 +16,1264 @@ const gems: Fig.Generator = {
...
+            description: "The platform of gem",
+          },
+        },
+        {
+          name: "--prerelease",
+          description: "Allow prerelease versions of a gem",
+        },
+        {
+          name: "--no-prerelease",
+          description: "Do not allow prerelease versions of a gem",
```
Comment (fedeci) [inline src/gem.ts:970]:
```suggestion
          description: "Do not allow prerelease versions of a gem",
          exclusiveOn: ["--prerelease"],
```

File: src/gem.ts [hunk 17]
```diff
@@ -16,18 +16,1264 @@ const gems: Fig.Generator = {
...
+          name: "--platform",
+          description: "Specify the platform of gem to specification",
+          args: {
+            name: "PLATFORM",
+            description: "The platform of gem",
+          },
+        },
+        {
+          name: "--prerelease",
+          description: "Allow prerelease versions of a gem",
```
Comment (fedeci) [inline src/gem.ts:966]:
```suggestion
          description: "Allow prerelease versions of a gem",
          exclusiveOn: ["--no-prerelease"],
```

File: src/gem.ts [hunk 18]
```diff
@@ -16,18 +16,1264 @@ const gems: Fig.Generator = {
...
+          description: "Remove all sources (clear the cache)",
+        },
+        {
+          name: ["-u", "--update"],
+          description: "Update source cache",
+        },
+        {
+          name: ["-f", "--force"],
+          description:
+            "Do not show any confirmation prompts and behave as if 'yes' was always answered",
```
Comment (fedeci) [inline src/gem.ts:922]:
```suggestion
            "Do not show any confirmation prompts and behave as if 'yes' was always answered",
          exclusiveOn: ["--no-force"],
```

File: src/gem.ts [hunk 19]
```diff
@@ -16,18 +16,1264 @@ const gems: Fig.Generator = {
...
+        },
+        {
+          name: "--no-force",
+          description:
+            "Show any confirmation prompts and behave as if 'yes' was always answered",
```
Comment (fedeci) [inline src/gem.ts:927]:
```suggestion
            "Show any confirmation prompts and behave as if 'yes' was always answered",
          exclusiveOn: ["--force"],
```

File: src/gem.ts [hunk 20]
```diff
@@ -16,18 +16,1264 @@ const gems: Fig.Generator = {
...
+          name: "-I",
+          description: "Equivalent to --no-installed",
+        },
+        {
+          name: ["-d", "--details"],
+          description: "Display detailed information of gem(s)",
+        },
+        {
+          name: "--no-details",
+          description: "Do not display detailed information of gem(s)",
```
Comment (fedeci) [inline src/gem.ts:849]:
```suggestion
          description: "Do not display detailed information of gem(s)",
          exclusiveOn: ["--details"],
```

File: src/gem.ts [hunk 21]
```diff
@@ -16,18 +16,1264 @@ const gems: Fig.Generator = {
...
+        isOptional: true,
+      },
+      options: [
+        {
+          name: "-I",
+          description: "Equivalent to --no-installed",
+        },
+        {
+          name: ["-d", "--details"],
+          description: "Display detailed information of gem(s)",
```
Comment (fedeci) [inline src/gem.ts:845]:
```suggestion
          description: "Display detailed information of gem(s)",
          exclusiveOn: ["--no-details"],
```

File: src/gem.ts [hunk 22]
```diff
@@ -16,18 +16,1264 @@ const gems: Fig.Generator = {
...
+      description: "Display remote gems whose name matches REGEXP",
+      args: {
+        name: "REGEXP",
+        description: "Regexp to look for in gem name",
+        isOptional: true,
+      },
+      options: [
+        {
+          name: "-I",
+          description: "Equivalent to --no-installed",
```
Comment (fedeci) [inline src/gem.ts:841]:
```suggestion
          description: "Equivalent to --no-installed",
          exclusiveOn: ["--installed"],
```

File: src/gem.ts [hunk 23]
```diff
@@ -16,18 +16,1264 @@ const gems: Fig.Generator = {
...
+          name: "--all",
+          description: "Generate RDoc/RI documentation for all installed gems",
+        },
+        {
+          name: "--rdoc",
+          description: "Generate RDoc HTML",
+        },
+        {
+          name: "--no-rdoc",
+          description: "Do not generate RDoc HTML",
```
Comment (fedeci) [inline src/gem.ts:802]:
```suggestion
          description: "Do not generate RDoc HTML",
          exclusiveOn: ["--rdoc"],
```

File: src/gem.ts [hunk 24]
```diff
@@ -16,18 +16,1264 @@ const gems: Fig.Generator = {
...
+        isOptional: true,
+      },
+      options: [
+        {
+          name: "--all",
+          description: "Generate RDoc/RI documentation for all installed gems",
+        },
+        {
+          name: "--rdoc",
+          description: "Generate RDoc HTML",
```
Comment (fedeci) [inline src/gem.ts:798]:
```suggestion
          description: "Generate RDoc HTML",
          exclusiveOn: ["--no-rdoc"],
```

File: src/gem.ts [hunk 25]
```diff
@@ -16,18 +16,1264 @@ const gems: Fig.Generator = {
...
+        },
+        {
+          name: "--no-rdoc",
+          description: "Do not generate RDoc HTML",
+        },
+        {
+          name: "--ri",
+          description: "Generate RI data",
```
Comment (fedeci) [inline src/gem.ts:806]:
```suggestion
          description: "Generate RI data",
          exclusiveOn: ["--no-ri"],
```

File: src/gem.ts [hunk 26]
```diff
@@ -16,18 +16,1264 @@ const gems: Fig.Generator = {
...
+        },
+        {
+          name: "--no-ri",
+          description: "Do not generate RI data",
```
Comment (fedeci) [inline src/gem.ts:810]:
```suggestion
          description: "Do not generate RI data",
          exclusiveOn: ["--ri"],
```

File: src/gem.ts [hunk 27]
```diff
@@ -16,18 +16,1264 @@ const gems: Fig.Generator = {
...
+        },
+        {
+          name: "--overwrite",
+          description: "Overwrite installed documents",
```
Comment (fedeci) [inline src/gem.ts:814]:
```suggestion
          description: "Overwrite installed documents",
          exclusiveOn: ["--no-overwrite"],
```

File: src/gem.ts [hunk 28]
```diff
@@ -16,18 +16,1264 @@ const gems: Fig.Generator = {
...
+        },
+        {
+          name: "--no-overwrite",
+          description: "Do not overwrite installed documents",
```
Comment (fedeci) [inline src/gem.ts:818]:
```suggestion
          description: "Do not overwrite installed documents",
          exclusiveOn: ["--overwrite"],
```

File: src/gem.ts [hunk 29]
```diff
@@ -16,18 +16,1264 @@ const gems: Fig.Generator = {
...
+          name: ["-n", "--name-matches"],
+          description: "Name of gem(s) to query on matches the provided REGEXP",
+          args: {
+            name: "REGEXP",
+            description: "The rule of the regexp",
+          },
+        },
+        {
+          name: "-I",
+          description: "Equivalent to --no-installed",
```
Comment (fedeci) [inline src/gem.ts:765]:
```suggestion
          description: "Equivalent to --no-installed",
          exclusiveOn: ["--installed"],
```

File: src/gem.ts [hunk 30]
```diff
@@ -16,18 +16,1264 @@ const gems: Fig.Generator = {
...
+        },
+        {
+          name: ["-d", "--details"],
+          description: "Display detailed information of gem(s)",
```
Comment (fedeci) [inline src/gem.ts:769]:
```suggestion
          description: "Display detailed information of gem(s)",
          exclusiveOn: ["--no-details"],
```

File: src/gem.ts [hunk 31]
```diff
@@ -16,18 +16,1264 @@ const gems: Fig.Generator = {
...
+        },
+        {
+          name: "--no-details",
+          description: "Do not display detailed information of gem(s)",
```
Comment (fedeci) [inline src/gem.ts:773]:
```suggestion
          description: "Do not display detailed information of gem(s)",
          exclusiveOn: ["--details"],
```

File: src/gem.ts [hunk 32]
```diff
@@ -16,18 +16,1264 @@ const gems: Fig.Generator = {
...
+          name: "--platform",
+          description: "Specify the platform of gem to fetch",
+          args: {
+            name: "PLATFORM",
+            description: "The platform of gem",
+          },
+        },
+        {
+          name: "--prerelease",
+          description: "Allow prerelease versions of a gem",
```
Comment (fedeci) [inline src/gem.ts:530]:
```suggestion
          description: "Allow prerelease versions of a gem",
          exclusiveOn: ["--no-prerelease"],
```

File: src/gem.ts [hunk 33]
```diff
@@ -16,18 +16,1264 @@ const gems: Fig.Generator = {
...
+        },
+        {
+          name: "--no-prerelease",
+          description: "Do not allow prerelease versions of a gem",
```
Comment (fedeci) [inline src/gem.ts:534]:
```suggestion
          description: "Do not allow prerelease versions of a gem",
          exclusiveOn: ["--prerelease"],
```

File: src/gem.ts [hunk 34]
```diff
@@ -16,18 +16,1264 @@ const gems: Fig.Generator = {
...
+        },
+        {
+          name: "--suggestions",
+          description: "Suggest alternates when gems are not found",
```
Comment (fedeci) [inline src/gem.ts:538]:
```suggestion
          description: "Suggest alternates when gems are not found",
          exclusiveOn: ["--no-suggestions"],
```

File: src/gem.ts [hunk 35]
```diff
@@ -16,18 +16,1264 @@ const gems: Fig.Generator = {
...
+        },
+        {
+          name: "--no-suggestions",
+          description: "Do not suggest alternates when gems are not found",
```
Comment (fedeci) [inline src/gem.ts:542]:
```suggestion
          description: "Do not suggest alternates when gems are not found",
          exclusiveOn: ["--suggestions"],
```

File: src/gem.ts [hunk 36]
```diff
@@ -16,18 +16,1264 @@ const gems: Fig.Generator = {
...
+          args: {
+            name: "DIRNAME",
+            template: "folders",
+            filterStrategy: "fuzzy",
+          },
+          requiresSeparator: true,
+        },
+        {
+          name: "--modern",
+          description: "Generate indexes for RubyGems (always true)",
```
Comment (fedeci) [inline src/gem.ts:584]:
```suggestion
          description: "Generate indexes for RubyGems (always true)",
          exclusiveOn: ["--no-modern"],
```

File: src/gem.ts [hunk 37]
```diff
@@ -16,18 +16,1264 @@ const gems: Fig.Generator = {
...
+        },
+        {
+          name: "--no-modern",
+          description: "Do not generate indexes for RubyGems",
```
Comment (fedeci) [inline src/gem.ts:588]:
```suggestion
          description: "Do not generate indexes for RubyGems",
          exclusiveOn: ["--modern"],
```

File: src/gem.ts [hunk 38]
```diff
@@ -16,18 +16,1264 @@ const gems: Fig.Generator = {
...
+        description: "Name and the version of gem to lock, ex: rails-1.0.0",
+      },
+      options: [
+        {
+          name: ["-s", "--strict"],
+          description: "Fail if unable to satisfy a dependency",
+        },
+        {
+          name: "--no-strict",
+          description: "Do not fail if unable to satisfy a dependency",
```
Comment (fedeci) [inline src/gem.ts:630]:
```suggestion
          description: "Do not fail if unable to satisfy a dependency",
          exclusiveOn: ["--strict"],
```

File: src/gem.ts [hunk 39]
```diff
@@ -16,18 +16,1264 @@ const gems: Fig.Generator = {
...
+      name: "lock",
+      description: "Generate a lockdown list of gems",
+      args: {
+        name: "GEMNAME-VERSION",
+        description: "Name and the version of gem to lock, ex: rails-1.0.0",
+      },
+      options: [
+        {
+          name: ["-s", "--strict"],
+          description: "Fail if unable to satisfy a dependency",
```
Comment (fedeci) [inline src/gem.ts:626]:
```suggestion
          description: "Fail if unable to satisfy a dependency",
          exclusiveOn: ["--no-strict"],
```

File: src/gem.ts [hunk 40]
```diff
@@ -16,18 +16,1264 @@ const gems: Fig.Generator = {
...
+          name: "--only-executables",
+          description: "Only restore executables",
+        },
+        {
+          name: "--only-plugins",
+          description: "Only restore plugins",
+        },
+        {
+          name: ["-E", "--env-shebang"],
+          description: "Rewrite executables with a shebang of /usr/bin/env",
```
Comment (fedeci) [inline src/gem.ts:715]:
```suggestion
          description: "Rewrite executables with a shebang of /usr/bin/env",
          exclusiveOn: ["--no-env-shebang"],
```

File: src/gem.ts [hunk 41]
```diff
@@ -16,18 +16,1264 @@ const gems: Fig.Generator = {
...
+        },
+        {
+          name: "--no-env-shebang",
+          description:
+            "Do not rewrite executables with a shebang of /usr/bin/env",
```
Comment (fedeci) [inline src/gem.ts:720]:
```suggestion
            "Do not rewrite executables with a shebang of /usr/bin/env",
          exclusiveOn: ["--env-shebang"],
```

File: src/gem.ts [hunk 42]
```diff
@@ -16,18 +16,1264 @@ const gems: Fig.Generator = {
...
+            generators: gems,
+            debounce: true,
+          },
+          requiresSeparator: true,
+          dependsOn: ["--all"],
+        },
+        {
+          name: "--extensions",
+          description:
+            "Restore gems with extensions in addition to regular gems",
```
Comment (fedeci) [inline src/gem.ts:698]:
```suggestion
            "Restore gems with extensions in addition to regular gems",
          exclusiveOn: ["--no-extensions"],
```

File: src/gem.ts [hunk 43]
```diff
@@ -16,18 +16,1264 @@ const gems: Fig.Generator = {
...
+        },
+        {
+          name: "--no-extensions",
+          description:
+            "Do not restore gems with extensions in addition to regular gems",
```
Comment (fedeci) [inline src/gem.ts:703]:
```suggestion
            "Do not restore gems with extensions in addition to regular gems",
          exclusiveOn: ["--extensions"],
```

**Review by fedeci — CHANGES_REQUESTED** [review]
Thanks @AlanSyue for this spec! It looks great, I can see it has been written surgically! I left a ton of suggestions but they are all the same, I think adding `exclusiveOn` will enhance this spec more!

**Review by fedeci — APPROVED** [review]
Thanks!

**Pull request merged**